qt-models/filtermodels.cpp

```diff
@@ -353,8 +354,10 @@ void LocationFilterModel::changeName(const QString &oldName, const QString &newN
 
 	// If there was already an entry with the new name, we are merging entries.
 	// Thus, if the old entry was selected, also select the new entry.
-	if (newIndex >= 0 && checkState[oldIndex])
+	if (newIndex >= 0 && checkState[oldIndex]) {
 		checkState[newIndex] = true;
+		emit dataChanged(createIndex(newIndex, 0), createIndex(newIndex, 0));
```

Comment:
Here also, there is first a setStringList() which to my reading of the docs does the right thing. Therefore, perhaps simply move this if-block in front of the setStringList().

Reply:
and agree.